bundles/core/src/test/java/com/adobe/cq/wcm/core/components/internal/link/DefaultPathProcessorTest.java

```diff
@@ -47,12 +53,28 @@ class DefaultPathProcessorTest {
     void testExternalizeWithException() {
         Externalizer externalizer = mock(Externalizer.class);
         when(externalizer.publishLink(any(ResourceResolver.class), anyString())).thenThrow(IllegalArgumentException.class);
-        localContext.registerService(externalizer);
-        DefaultPathProcessor underTest = localContext.registerService(new DefaultPathProcessor());
+        localContext.registerService(Externalizer.class, externalizer);
+        DefaultPathProcessor underTest = localContext.registerInjectActivateService(new DefaultPathProcessor());
         String path = PATH;
         assertEquals(path, underTest.externalize(path, localContext.request()));
     }
 
+    @Test
+    void testExternalizerIsCalledWithMaskedUrl() {
+        ArgumentCaptor<String> url = ArgumentCaptor.forClass(String.class);
+        Externalizer externalizer = mock(Externalizer.class);
+        when(externalizer.publishLink(any(ResourceResolver.class), anyString())).then(AdditionalAnswers.returnsSecondArg());
+        localContext.registerService(Externalizer.class, externalizer);
+        DefaultPathProcessor underTest = localContext.registerInjectActivateService(new DefaultPathProcessor());
+        String path = PATH + ".html?recipient=<%= recipient.id %>";
+
+        assertEquals(path, underTest.externalize(path, localContext.request()));
+
+        verify(externalizer).publishLink(any(ResourceResolver.class), url.capture());
+        assertFalse(url.getValue().contains("<%="));
```

Comment:
I would feel more confident if we asserted on the expected output and not on simply not having the original fragments. This would also catch situation where we might end up up with multiple masking/encoding.